### ISSUE TITLE
Add the menu option to set Translation source

### ISSUE THREAD

**Issue by angrezichatterbox** [scribe-org/Scribe-Android#149]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Currently the individual language settings doesn't allow the user to set the translation source for each language keyboard settings. The issue aims to add the ability to set translation source language for each language keyboard separately. This could be stored in the shared preferences for now for each language separately similar to how it is being done for period_on_double_tap. 

 A visual reference of how this option would look is given below. 
 
 <img width="385" alt="Screenshot 2024-09-24 at 12 39 00 AM" src="https://github.com/user-attachments/assets/46d55d8d-db09-4472-af74-29133ffc7a49">

### Contribution

I would love to work on this and am more than happy to help anyone fix this issue :)

**Comment by ThousandYearGrind** [comment]
@henrikth39, Hi, does this work?

**Comment by andrewtavis** [comment]
Hey @ThousandYearGrind 👋 Quick note that this issue is now just to add the menu option that selects from a list, so at the end of it we want the menu option that's shown in the issue text and the interface like [this in the Figma designs](https://www.figma.com/design/c8945w2iyoPYVhsqW7vRn6/scribe_public_designs?node-id=544-1527&t=xkn0oeQkAyfkQzxU-1). Actually switching the translation language will be handled by #265 once this issue is finished - i.e. this issue is for the interface and #265 is for the functionality.

**Comment by ThousandYearGrind** [comment]
Ahhh, thank you. I was scratching my head over that for a while.

**Comment by ThousandYearGrind** [comment]
@andrewtavis How should I approach translations? Can I just run "Translation source language" through google translate or should this be done later?

**Comment by andrewtavis** [comment]
We'll have local translations for this, @ThousandYearGrind :) This issue is just for the menu option to select the base language.

**Comment by ThousandYearGrind** [comment]
Hi @andrewtavis, where would be a good place to ask questions about code-specific questions? I've been trying to figure out how to do something for a while.

**Comment by andrewtavis** [comment]
@ThousandYearGrind, asking them here would be good, that way it's easy for us to link to files :)

**Comment by ThousandYearGrind** [comment]
So I've added the option for the settings screen using a ClickableItem like the app language setting. Now, I need to set the action that takes place when the user clicks on the setting.

`val appSettingsItemList =
        listOf(
            ScribeItem.ClickableItem(
                title = stringResource(R.string.app_settings_menu_app_language),
                desc = stringResource(R.string.app_settings_menu_app_language_description),
                action = onLanguageSelect,
            ),
            ScribeItem.SwitchItem(
                title = stringResource(R.string.app_settings_menu_app_color_mode),
                desc = stringResource(R.string.app_settings_menu_app_color_mode_description),
                state = isUserDarkMode,
                onToggle = { isUserDarkMode1 ->
                    onDarkModeChange(isUserDarkMode1)
                },
            ),
            ScribeItem.SwitchItem(
                title = stringResource(R.string.app_settings_keyboard_keypress_vibration),
                desc = stringResource(R.string.app_settings_keyboard_keypress_vibration_description),
                state = vibrateOnKeypress.value,
                onToggle = { shouldVibrateOnKeypress ->
                    vibrateOnKeypress.value = shouldVibrateOnKeypress
                    PreferencesHelper.setVibrateOnKeypress(context, shouldVibrateOnKeypress)
                },
            ),
            ScribeItem.SwitchItem(
                title = stringResource(R.string.app_settings_keyboard_functionality_popup_on_keypress),
                desc = stringResource(R.string.app_settings_keyboard_functionality_popup_on_keypress_description),
                state = popupOnKeypress.value,
                onToggle = { shouldPopUpOnKeypress ->
                    popupOnKeypress.value = shouldPopUpOnKeypress
                    PreferencesHelper.setShowPopupOnKeypress(context, shouldPopUpOnKeypress)
                },
            ),
            ScribeItem.ClickableItem(
                title = stringResource(R.string.app_settings_menu_translation_language),
                desc = stringResource(R.string.app_settings_menu_translation_language_description),
                action = onTranslationLanguageSelect,
            ),
        )`

<img width="361" alt="Screenshot 2024-12-21 at 4 40 15 PM" src="https://github.com/user-attachments/assets/38cac7c9-39d1-4796-a9ce-268a215adf20" />

I looked at the app language setting's action and it uses an intent, but I'm not sure how to do this for the translation language setting.

`private fun selectTranslationLanguage() {
        
    }

    private fun selectLanguage() {
        val packageName = requireActivity().packageName
        val intent =
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
                Intent(ACTION_APP_LOCALE_SETTINGS)
            } else {
                Intent(ACTION_APPLICATION_DETAILS_SETTINGS).apply {
                    data = Uri.fromParts("package", packageName, null)
                    putExtra(
                        "android.intent.extra.SHOW_FRAGMENT",
                        "com.android.settings.localepicker.LocaleListEditor",
                    )
                }
            }
        intent.data = Uri.fromParts("package", packageName, null)
        startActivity(intent)
    }`

**Comment by andrewtavis** [comment]
Hey @ThousandYearGrind 👋 Firstly, could we get this menu option moved into the settings for each language keyboard, so it wouldn't be on settings, but on your screen it would be in French and English. We made the decision to have it there as there are members of the community who would translate from different languages: say French to Spanish and English to German. 

From there we won't be having it function like the app's language selection. Clicking that option will take the user to a sub menu with radio options. Maybe you can make the change for the menu option location and then open a PR and we can then assist on making the radio option menu?

**Comment by ThousandYearGrind** [comment]
@andrewtavis I've done as you suggested and moved the menu option to the language keyboard. Should I submit a PR now?                         
<img width="304" alt="Screenshot 2024-12-27 at 3 41 29 PM" src="https://github.com/user-attachments/assets/dd558b1d-e714-478e-940d-5ac3f0d21b49" />

**Comment by andrewtavis** [comment]
We're basically there, @ThousandYearGrind! Could you refer to where it should be located in [the designs for this tab](https://www.figma.com/design/c8945w2iyoPYVhsqW7vRn6/scribe_public_designs?node-id=513-1616&t=zv0JMRYMzn0jxAME-1)? Pasting an image of its location in its own sub menu below as well:

![Keyboard Settings - Light](https://github.com/user-attachments/assets/6696005b-e91a-48bd-ba8f-7c7c2f282ac7)

Feel free to open a PR once it's been moved!

**Comment by andrewtavis** [comment]
And thanks for your persistence here!

**Comment by ThousandYearGrind** [comment]
Hi @andrewtavis,
Sorry for the delay. I've had quite a lot of things come up IRL, so I've been neglecting my work on this project a bit. However, I put in some work today and finished adding the menu option like you said. Here is what it looks like now:
<img width="313" alt="Screenshot 2025-01-15 at 1 25 02 PM" src="https://github.com/user-attachments/assets/6542a0bd-ea24-4b8b-9507-30352318bd78" />
I'm clear to submit the PR, correct?

**Comment by andrewtavis** [comment]
Yes definitely, @ThousandYearGrind! And no stress on it taking a bit :) Thanks for the great work!

**Comment by angrezichatterbox** [comment]
Hey @ThousandYearGrind, hope you're doing well! Just checking in—we're good to go for a PR with the changes you shared. Looking forward to seeing your PR soon!

**Comment by ThousandYearGrind** [comment]
I must've forgotten to do that, I'll do it soon.

**Comment by ThousandYearGrind** [comment]
I'm running tests and I get this error message:
`Invalid value (23) passed to --jvm-target, must be one of [1.6, 1.8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21]`
when i run:
`./gradlew lintKotlin detekt`

**Comment by angrezichatterbox** [comment]
Detekt is not compatible with Java version 23. It is only compatible up to Java version 21. Could you try installing Java 17 and check if it works?

**Comment by angrezichatterbox** [comment]
Hey @ThousandYearGrind Were you able to run detekt on your device. Happy to help in case you are facing difficulties.

**Comment by ThousandYearGrind** [comment]
> Hey @ThousandYearGrind Were you able to run detekt on your device. Happy to help in case you are facing difficulties. 

I'll try to do it this weekend, currently quite busy.

**Comment by ThousandYearGrind** [comment]
Made some progress and managed to remove my java 23. Now I had some issues merging so I'll be working on transferring my stuff to a clone of the current repository.

**Comment by andrewtavis** [comment]
Great to hear about the progress here, @ThousandYearGrind! Please let us know if there's something we can do to support 😊 Looking forward to the PR!

**Comment by ThousandYearGrind** [comment]
I'm going preparing for a hackathon on Saturday which is why I haven't submitted the pull request yet; I should send it in sometime on Sunday.

**Comment by andrewtavis** [comment]
Thanks for the update, @ThousandYearGrind, and have a great time at the hackathon! :)

**Comment by ThousandYearGrind** [comment]
Pull request submitted

**Comment by andrewtavis** [comment]
Awesome, @ThousandYearGrind! Thanks for it :) We'll try to get to the review soon 😊

**Comment by andrewtavis** [comment]
Closed by #330 😊 Thanks for the great work here, @ThousandYearGrind and @angrezichatterbox! Another amazing step towards parity with the iOS app as this isn't even released yet for iOS 🚀

**Comment by ThousandYearGrind** [comment]
Thank you @andrewtavis 
This was my first venture into coding outside a classroom environment and I've learned a lot from it!

**Comment by andrewtavis** [comment]
Feel free to pick up another issue in the future, @ThousandYearGrind! :)
